### PR TITLE
logging: use kind-prefixed fields instead of string-formatting

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Part of #1694 
/assign @sttts 

Using the resource instead of the kind would be cool but I don't know that there's any lightweight way to figure that out.

### CODE REVIEW

File: pkg/logging/constants.go [hunk 1]
```diff
@@ -52,28 +54,28 @@ func WithQueueKey(logger logr.Logger, key string) logr.Logger {
...
 // WithObject adds object identifiers to the logger.
-func WithObject(logger logr.Logger, obj metav1.Object) logr.Logger {
+func WithObject(logger logr.Logger, obj Object) logr.Logger {
 	return logger.WithValues(From(obj)...)
 }
 
 // From provides the structured logging fields that identify an object.
-func From(obj metav1.Object) []interface{} {
+func From(obj Object) []interface{} {
+	gvk := obj.GetObjectKind().GroupVersionKind()
```
Comment (sttts) [inline pkg/logging/constants.go:69]: and if empty, then `%T` ?

File: pkg/reconciler/apis/apibinding/apibinding_controller.go [hunk 2]
```diff
@@ -281,9 +281,8 @@ func (c *controller) enqueueCRD(obj interface{}, logger logr.Logger) {
 		runtime.HandleError(fmt.Errorf("obj is supposed to be a CustomResourceDefinition, but is %T", obj))
 		return
 	}
-	logger = logger.WithValues(
-		"CustomResource", fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group),
-		"CustomResourceDefinition", logging.Key(crd),
+	logger = logging.WithObject(logger, crd).WithValues(
+		"groupResource", fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group),
```
Comment (sttts) [inline pkg/reconciler/apis/apibinding/apibinding_controller.go:285]: gr

File: pkg/logging/constants.go [hunk 3]
```diff
@@ -40,6 +40,8 @@ const (
 	NamespaceKey = "namespace"
 	// NameKey is used to specify a name when a log is related to an object.
 	NameKey = "name"
+	// APIVersionKey is used to specify an API version when a log is related to an object.
+	APIVersionKey = "apiVersion"
```
Comment (sttts) [inline pkg/logging/constants.go:44]: I think we short find short keys for the common ones. `ws`, `ns`, `v`

File: pkg/logging/constants.go [hunk 4]
```diff
@@ -52,28 +54,28 @@ func WithQueueKey(logger logr.Logger, key string) logr.Logger {
...
-func WithObject(logger logr.Logger, obj metav1.Object) logr.Logger {
+func WithObject(logger logr.Logger, obj Object) logr.Logger {
 	return logger.WithValues(From(obj)...)
 }
 
 // From provides the structured logging fields that identify an object.
-func From(obj metav1.Object) []interface{} {
+func From(obj Object) []interface{} {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	prefix := strings.ToLower(gvk.Kind[0:1]) + gvk.Kind[1:]
```
Comment (sttts) [inline pkg/logging/constants.go:70]: then make it completely lowercase.

Comment (sttts) [reply pkg/logging/constants.go:70]: Not configMap.

File: pkg/logging/constants.go (hunk 1 above)
Comment (stevekuznetsov) [reply pkg/logging/constants.go:69]: Good idea!

File: pkg/reconciler/apis/apibinding/apibinding_controller.go (hunk 2 above)
Comment (stevekuznetsov) [reply pkg/reconciler/apis/apibinding/apibinding_controller.go:285]: verbosity > terseness

File: pkg/logging/constants.go (hunk 3 above)
Comment (stevekuznetsov) [reply pkg/logging/constants.go:44]: I generally disagree ;)

File: pkg/logging/constants.go (hunk 4 above)
Comment (stevekuznetsov) [reply pkg/logging/constants.go:70]: ACK!

**Comment by stevekuznetsov** [comment]
@sttts updated - re: short-names ... any UI (even the CLI ones) for filtering structured logs do autocomplete for you, and I prefer the readability and explicitness of longer names, for variables, flags, etc. If we find ourselves manually typing something a lot I'd investigate why, then maybe fix that, _then_ look into shorter names. We furthermore have the ability to map names to shorter things if we're developing a viewing layer on top anyway.

**Comment by sttts** [comment]
Let's merge and learn whether this is what we want.

/lgtm
/approve

**Comment by stevekuznetsov** [comment]
wow. unit test flake

/retest

**Pull request merged**